protocolapi/block/api.go

```diff
@@ -0,0 +1,160 @@
+package block
```

Comment:
i think it would be simplest to have the api live in the same package as the protocol. however it seems like something is preventing that ("When I put it into the protocol directory it had problems recognizing where the api functions were"), but i'm having trouble understanding what the problem is. can you help me understand?